enterprise/server/remote_execution/snaploader/snaploader_test.go

```diff
@@ -89,6 +89,66 @@ func TestPackAndUnpackChunkedFiles(t *testing.T) {
 	require.NoError(t, err)
 	workDir := testfs.MakeTempDir(t)
 
+	// Create an initial chunked file for VM A.
+	workDirA := testfs.MakeDirAll(t, workDir, "VM-A")
+	const chunkSize = 512 * 1024
+	const fileSize = 13 + (chunkSize * 10) // ~5 MB total, with uneven size
+	originalImagePath := makeRandomFile(t, workDirA, "scratchfs.ext4", fileSize)
+	cowA, err := copy_on_write.ConvertFileToCOW(env.GetFileCache(), originalImagePath, chunkSize, workDirA)
+	require.NoError(t, err)
+
+	// Overwrite a random range to simulate the disk being written to. This
+	// should create some dirty chunks.
+	writeRandomRange(t, cowA)
+
+	// Now store a snapshot for VM A, including the COW we created.
+	task := &repb.ExecutionTask{}
+	key, err := snaploader.NewKey(task, "config-hash", "")
+	require.NoError(t, err)
+	optsA := makeFakeSnapshot(t, workDirA)
+	optsA.ChunkedFiles = map[string]*copy_on_write.COWStore{
+		"scratchfs": cowA,
+	}
+	snapA, err := loader.CacheSnapshot(ctx, key, optsA)
+	require.NoError(t, err)
+	// Note: we'd normally close cowA here, but we keep it open so that
+	// mustUnpack() can verify the original contents.
+
+	// We want to make sure we can save/unpack snapshots that were also
+	// unpacked from a snapshot.
+	// i.e. For SnapA -> ForkA -> ForkA' we want to make sure ForkA' functions
+	// correctly
+	originalSnap := snapA
```

Comment:
This differs from the pre-existing unit test because we are testing unpacking and saving a snapshot that was unpacked from another snapshot. 

For SnapA -> ForkA -> ForkA':
                   -> ForkB
The existing test checks that ForkA and ForkB don't clobber each other.
This test checks that ForkA' works as expected